src/bin/add/main.rs

```diff
@@ -143,6 +147,34 @@ fn handle_add(args: &Args) -> Result<()> {
     }
     let deps = &args.parse_dependencies()?;
 
+    if let Some(ref requested_features) = args.features {
+        let available_features = deps
+            .iter()
+            .map(|d| d.available_features.clone())
+            .flatten()
+            .collect::<Vec<String>>();
+        let flattened_features: Vec<String> = requested_features
+            .iter()
+            .map(|feat| {
+                feat.split(' ')
+                    .map(|s| s.trim().to_string())
+                    .collect::<Vec<String>>()
+            })
+            .flatten()
+            .collect();
+        let unknown_features: Vec<String> = flattened_features
+            .iter()
+            .filter(|req_feat| !available_features.contains(req_feat) && !req_feat.is_empty())
```

Comment:
Nit: If we use a `BTreeSet` for each of these, instead of a `Vec`, we can have it find the differences for us and keep them sorted.

Reply:
done :)